quinn-proto/src/connection/mod.rs

```diff
@@ -760,6 +761,18 @@ where
                     self.path.challenge_pending = false;
                 }
                 Timer::Pacing => trace!("pacing timer expired"),
+                Timer::PushNewCid => {
+                    // Update `retire_prior_to` field in NEW_CONNECTION_ID frame
+                    let num_new_cid = self.local_cid_state.on_cid_retirement_timeout().into();
+                    if !self.state.is_closed() {
+                        trace!(
+                            "push a new cid to peer RETIRE_PRIOR_TO field {}",
+                            self.local_cid_state.retire_seq
+                        );
+                        self.endpoint_events
+                            .push_back(EndpointEventInner::NeedIdentifiers(now, num_new_cid));
+                    }
```

Comment:
This could also be moved around a little to make `CidState` be more of an abstraction.

Reply:
Related logic is already moved into `on_cid_retirement_timeout`. I don't see more we can move to increase abstraction here.

Reply:
I guess I was wondering about the `trace!()` call here, but that might not be practical.